apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -80,49 +80,40 @@ export const commands = {
   // current or previous frame.
   anySpeechIncludesValues(currentVariables, previousVariables) {
     const spriteIds = this.getSpriteIdsInUse();
-    let result = false;
-    const values = previousVariables
-      ? Object.values(currentVariables).concat(Object.values(previousVariables))
-      : Object.values(currentVariables);
-    for (let i = 0; i < spriteIds.length; i++) {
-      values.forEach(value => {
-        const speechText = this.getLastSpeechBubbleForSpriteId(
-          spriteIds[i]
-        )?.text;
-        const type = typeof speechText;
-        // We only want to set result to true here, so that any positive test
-        // allows the overall criterion to pass.
-        switch (type) {
-          case 'string':
-            if (speechText.includes(value)) {
-              result = true;
-            }
-            break;
-          case 'number':
-            if (speechText === value) {
-              result = true;
-            }
-            break;
-          default:
-            break;
+    const values = Object.values(currentVariables);
+    if (previousVariables) {
+      values.push(...Object.values(previousVariables));
```

Comment:
I find this use of push slightly unusual, because it's pushing the variables into the first array as individual elements. For a small array like this, it doesn't matter, but if you're ever dealing with really large arrays, I think there might be performance issues with using `.push()` like this vs. the original `.concat()`. 

You could also do:
```
const values = [
  ...Object.values(currentVariables),
  ...(previousVariables ? Object.values(previousVariables) : [])
];
```

or

```
const values = Object.values(currentVariables).concat(previousVariables ? Object.values(previousVariables) : []);
```

Reply:
I'll switch to `.concat`. I wasn't too worried about performance because I wouldn't expect most students to have more than a few variables.